fw/http_parser.c

```diff
@@ -8600,45 +8715,134 @@ __resp_parse_cache_control(TfwHttpResp *resp, unsigned char *data, size_t len)
 	__FSM_STATE(Resp_I_CC_n) {
 		TRY_STR_LAMBDA_fixup(&TFW_STR_STRING("no-cache"),
 			&parser->hdr, {
-			parser->_acc = TFW_HTTP_CC_NO_CACHE;
-		}, Resp_I_CC_n, Resp_I_Flag);
+			parser->cc_dir_flag = TFW_HTTP_CC_NO_CACHE;
+		}, Resp_I_CC_n, Resp_I_Dir_Eq);
 		TRY_STR_LAMBDA_fixup(&TFW_STR_STRING("no-store"),
 			&parser->hdr, {
-			parser->_acc = TFW_HTTP_CC_NO_STORE;
+			parser->cc_dir_flag = TFW_HTTP_CC_NO_STORE;
 		}, Resp_I_CC_n, Resp_I_Flag);
 		TRY_STR_LAMBDA_fixup(&TFW_STR_STRING("no-transform"),
 			&parser->hdr, {
-			parser->_acc = TFW_HTTP_CC_NO_TRANSFORM;
+			parser->cc_dir_flag = TFW_HTTP_CC_NO_TRANSFORM;
 		}, Resp_I_CC_n, Resp_I_Flag);
 		TRY_STR_INIT();
 		__FSM_I_JMP(Resp_I_Ext);
 	}
 
 	__FSM_STATE(Resp_I_CC_p) {
 		TRY_STR_LAMBDA_fixup(&TFW_STR_STRING("public"), &parser->hdr, {
-			parser->_acc = TFW_HTTP_CC_PUBLIC;
+			parser->cc_dir_flag = TFW_HTTP_CC_PUBLIC;
 		}, Resp_I_CC_p, Resp_I_Flag);
 		TRY_STR_LAMBDA_fixup(&TFW_STR_STRING("private"), &parser->hdr, {
-			parser->_acc = TFW_HTTP_CC_PRIVATE;
-		}, Resp_I_CC_p, Resp_I_Flag);
+			parser->cc_dir_flag = TFW_HTTP_CC_PRIVATE;
+		}, Resp_I_CC_p, Resp_I_Dir_Eq);
 		TRY_STR_LAMBDA_fixup(&TFW_STR_STRING("proxy-revalidate"),
 			&parser->hdr, {
-			parser->_acc = TFW_HTTP_CC_PROXY_REVAL;
+			parser->cc_dir_flag = TFW_HTTP_CC_PROXY_REVAL;
 		}, Resp_I_CC_p, Resp_I_Flag);
 		TRY_STR_INIT();
 		__FSM_I_JMP(Resp_I_Ext);
 	}
 
+	/*
+	 * no-cache or private with arguments as defined by rfc7234
+	 * sections 5.2.2.2 and 5.2.2.6
+	 */
+	__FSM_STATE(Resp_I_Dir_Eq) {
+		if (c == '=') {
+			/* Qualified form cancels out the general flag */
+			/*
+			 * Resp_I_CC_Dir_Opening_Quote =>
+			 * => Resp_I_CC_Dir_Arg_Token =>
+			 * => Resp_I_CC_Dir_Arg_EoT
+			 */
+			__FSM_I_MOVE_fixup(Resp_I_CC_Dir_Opening_Quote, 1, 0);
+		}
+		__FSM_I_JMP(Resp_I_Flag);
+	}
+
 	__FSM_STATE(Resp_I_Flag) {
-		WARN_ON_ONCE(!parser->_acc);
+		/* A start of a standard directive successfully detected */
 		if (IS_WS(c) || c == ',' || IS_CRLF(c)) {
-			resp->cache_ctl.flags |= parser->_acc;
-			parser->_acc = 0;
+			resp->cache_ctl.flags |= parser->cc_dir_flag;
 			__FSM_I_JMP(Resp_I_EoT);
 		}
+		/* ...but the flag appears to have an unknown suffix */
 		__FSM_I_JMP(Resp_I_Ext);
 	}
 
+	__FSM_STATE(Resp_I_CC_Dir_Opening_Quote) {
+		if (c != '"')
+			__FSM_EXIT(TFW_BLOCK);
+
+		__FSM_I_MOVE_fixup(Resp_I_CC_Dir_Arg_Token, 1, 0);
+	}
+
+	__FSM_STATE(Resp_I_CC_Dir_Arg_Token) {
+		/* comma-separated field list (field-name/token) with optional
+		 * linear white space
```

Comment:
Please start the sentence form Captital letter and end it with dot.

Also please add following comment here: "We do not set TfwStr->skb for tokens, since the tokens are used in read-only fashion to compare strings and are never rewriten (TfwStr->skb is used for rewriting raw data only)."